### PR TITLE
fix: make history local to a given excalidraw instance

### PR DESCRIPTION
**Pull request by ad1992**
An attempt to make history local to excalidraw instance so each component has its own history when [multiple excalidraw](https://github.com/excalidraw/excalidraw/issues/3043) rendered.

Try [here](https://k1xx5.csb.app/)

### CODE REVIEW

**Comment by dwelle** [comment]
Nice. What's up with the snapshot changes?

**Comment by dwelle** [comment]
:+1: for me.

PR:

image

production: 

image

**Review by dwelle — APPROVED** [review]

**Comment by ad1992** [comment]
> Nice. What's up with the snapshot changes?



> Nice. What's up with the snapshot changes?

I have been checking about the same but looks like cleanup not happening correctly due to shared history, the only difference between the snapshots is now the snapshot with `width 0` is removed but not sure why it was there earlier though.

File: src/packages/excalidraw/CHANGELOG.md [hunk 1]
```diff
@@ -38,6 +38,8 @@ Please add the latest change on the top under the correct section.
 
 ### Fixes
 
+- Make history local to a given excalidraw instance. This fixes a case where history was getting shared when you have multiple Excalidraw components on same page [#3481](https://github.com/excalidraw/excalidraw/pull/3481).
```
Comment (ad1992) [inline src/packages/excalidraw/CHANGELOG.md:41]:
```suggestion
- Make history local to a given Excalidraw instance. This fixes a case where history was getting shared when you have multiple Excalidraw components on the same page [#3481](https://github.com/excalidraw/excalidraw/pull/3481).
```

**Pull request merged**